### PR TITLE
Compute and Convert classes of Revit_Core_Engine covered with descriptions

### PR DESCRIPTION
**Pull request by pawelbaran**
### Issues addressed by this PR

1st part of #596

### Test files

This PR is only about descriptions.

### Additional comments

In the coming days, this PR will be followed with sister PRs covering other classes of Revit_Core_Engine.

### CODE REVIEW

**Comment by pawelbaran** [comment]
@BHoMBot check compliance

**Comment by pawelbaran** [comment]
@BHoMBot check compliance

File: Revit_Core_Engine/Compute/LoadFamilySymbol.cs [hunk 1]
```diff
@@ -34,6 +36,13 @@ public static partial class Compute
         /****              Public methods               ****/
         /***************************************************/
 
+        [Description("Loads the Revit family type of given category and names into a given document.")]
+        [Input("familyLoadSettings", "Family load settings object containing the family library to be searched.")]
+        [Input("document", "Revit document to load the family type into.")]
+        [Input("categoryName", "Category, to which the sought family belongs.")]
+        [Input("familyName", "Name of the sought family.")]
+        [Input("familyTypeName", "Name of the sought family type.")]
+        [Output("familyType", "Family type loaded to the Revit document based on the input criteria and settings.")]
```
Comment (tiagogrossi) [inline Revit_Core_Engine/Compute/LoadFamilySymbol.cs:45]: Should be `familySymbol`?

File: Revit_Core_Engine/Convert/Geometry/ToRevit/Grid.cs [hunk 2]
```diff
@@ -36,6 +37,12 @@ public static partial class Convert
         /****              Public methods               ****/
         /***************************************************/
 
+        [Description("Converts BH.oM.Geometry.SettingOut.Grid to a Revit Grid or MultiSegmentGrid.")]
+        [Input("grid", "BH.oM.Geometry.SettingOut.Grid to be converted.")]
+        [Input("document", "Revit document, in which the output of the convert will be created.")]
+        [Input("settings", "Revit adapter settings to be used while performing the convert.")]
+        [Input("refObjects", "Optional, a collection of objects already processed in the current adapter action, stored to avoid processing the same object more than once.")]
+        [Output("grid", "Revit Grid or MultiSegmentGrid resulting from converting the input BH.oM.Geometry.SettingOut.Grid.")]
```
Comment (tiagogrossi) [inline Revit_Core_Engine/Convert/Geometry/ToRevit/Grid.cs:45]: Should be `element`? Returns grid as an Element.

File: Revit_Core_Engine/Convert/Physical/ToRevit/FamilyInstance.cs [hunk 3]
```diff
@@ -37,6 +39,12 @@ public static partial class Convert
         /****            Interface methods              ****/
         /***************************************************/
 
+        [Description("Converts BH.oM.Physical.Elements.IFramingElement to a Revit FamilyInstance.")]
+        [Input("framingElement", "BH.oM.Physical.Elements.IFramingElement to be converted.")]
+        [Input("document", "Revit document, in which the output of the convert will be created.")]
+        [Input("settings", "Revit adapter settings to be used while performing the convert.")]
+        [Input("refObjects", "Optional, a collection of objects already processed in the current adapter action, stored to avoid processing the same object more than once.")]
+        [Output("instance", "Revit FamilyInstance resulting from converting the input BH.oM.Physical.Elements.IFramingElement.")]
```
Comment (tiagogrossi) [inline Revit_Core_Engine/Convert/Physical/ToRevit/FamilyInstance.cs:47]: Should be `familyInstance`?

File: Revit_Core_Engine/Convert/Physical/ToRevit/FamilyInstance.cs [hunk 4]
```diff
@@ -47,6 +55,12 @@ public static FamilyInstance IToRevitFamilyInstance(this IFramingElement framing
         /****              Public methods               ****/
         /***************************************************/
 
+        [Description("Converts BH.oM.Physical.Elements.Column to a Revit FamilyInstance.")]
+        [Input("framingElement", "BH.oM.Physical.Elements.Column to be converted.")]
+        [Input("document", "Revit document, in which the output of the convert will be created.")]
+        [Input("settings", "Revit adapter settings to be used while performing the convert.")]
+        [Input("refObjects", "Optional, a collection of objects already processed in the current adapter action, stored to avoid processing the same object more than once.")]
+        [Output("instance", "Revit FamilyInstance resulting from converting the input BH.oM.Physical.Elements.Column.")]
```
Comment (tiagogrossi) [inline Revit_Core_Engine/Convert/Physical/ToRevit/FamilyInstance.cs:63]: Should be `familyInstance`?

File: Revit_Core_Engine/Convert/Physical/ToRevit/FamilyInstance.cs [hunk 5]
```diff
@@ -137,6 +151,12 @@ public static FamilyInstance ToRevitFamilyInstance(this Column framingElement, D
 
         /***************************************************/
 
+        [Description("Converts BH.oM.Physical.Elements.Pile to a Revit FamilyInstance.")]
+        [Input("framingElement", "BH.oM.Physical.Elements.Pile to be converted.")]
+        [Input("document", "Revit document, in which the output of the convert will be created.")]
+        [Input("settings", "Revit adapter settings to be used while performing the convert.")]
+        [Input("refObjects", "Optional, a collection of objects already processed in the current adapter action, stored to avoid processing the same object more than once.")]
+        [Output("instance", "Revit FamilyInstance resulting from converting the input BH.oM.Physical.Elements.Pile.")]
```
Comment (tiagogrossi) [inline Revit_Core_Engine/Convert/Physical/ToRevit/FamilyInstance.cs:159]: Should be `familyInstance`?

File: Revit_Core_Engine/Convert/Physical/ToRevit/FamilyInstance.cs [hunk 6]
```diff
@@ -145,6 +165,12 @@ public static FamilyInstance ToRevitFamilyInstance(this Pile framingElement, Doc
 
         /***************************************************/
 
+        [Description("Converts BH.oM.Physical.Elements.IFramingElement to a Revit FamilyInstance.")]
+        [Input("framingElement", "BH.oM.Physical.Elements.IFramingElement to be converted.")]
+        [Input("document", "Revit document, in which the output of the convert will be created.")]
+        [Input("settings", "Revit adapter settings to be used while performing the convert.")]
+        [Input("refObjects", "Optional, a collection of objects already processed in the current adapter action, stored to avoid processing the same object more than once.")]
+        [Output("instance", "Revit FamilyInstance resulting from converting the input BH.oM.Physical.Elements.IFramingElement.")]
```
Comment (tiagogrossi) [inline Revit_Core_Engine/Convert/Physical/ToRevit/FamilyInstance.cs:173]: Should be `familyInstance`?

File: Revit_Core_Engine/Convert/Revit/ToRevit/CurveElement.cs [hunk 7]
```diff
@@ -39,7 +41,13 @@ public static partial class Convert
         /****               Public Methods              ****/
         /***************************************************/
 
-        public static Element ToCurveElement(this ModelInstance modelInstance, Document document, RevitSettings settings = null, Dictionary<Guid, List<int>> refObjects = null)
+        [Description("Converts BH.oM.Adapters.Revit.Elements.ModelInstance to a Revit CurveElement.")]
+        [Input("modelInstance", "BH.oM.Adapters.Revit.Elements.ModelInstance to be converted.")]
+        [Input("document", "Revit document, in which the output of the convert will be created.")]
+        [Input("settings", "Revit adapter settings to be used while performing the convert.")]
+        [Input("refObjects", "Optional, a collection of objects already processed in the current adapter action, stored to avoid processing the same object more than once.")]
+        [Output("element", "Revit CurveElement resulting from converting the input BH.oM.Adapters.Revit.Elements.ModelInstance.")]
```
Comment (tiagogrossi) [inline Revit_Core_Engine/Convert/Revit/ToRevit/CurveElement.cs:49]: Should be `curveElement`?

File: Revit_Core_Engine/Convert/Revit/ToRevit/CurveElement.cs [hunk 8]
```diff
@@ -94,6 +102,12 @@ public static Element ToCurveElement(this ModelInstance modelInstance, Document
 
         /***************************************************/
 
+        [Description("Converts BH.oM.Adapters.Revit.Elements.DraftingInstance to a Revit CurveElement.")]
+        [Input("draftingInstance", "BH.oM.Adapters.Revit.Elements.DraftingInstance to be converted.")]
+        [Input("document", "Revit document, in which the output of the convert will be created.")]
+        [Input("settings", "Revit adapter settings to be used while performing the convert.")]
+        [Input("refObjects", "Optional, a collection of objects already processed in the current adapter action, stored to avoid processing the same object more than once.")]
+        [Output("element", "Revit CurveElement resulting from converting the input BH.oM.Adapters.Revit.Elements.DraftingInstance.")]
```
Comment (tiagogrossi) [inline Revit_Core_Engine/Convert/Revit/ToRevit/CurveElement.cs:110]: Should be `curveElement`?

File: Revit_Core_Engine/Convert/ToRevit.cs [hunk 9]
```diff
@@ -38,41 +40,77 @@ public static partial class Convert
         /****              Public methods               ****/
         /***************************************************/
 
+        [Description("Converts BH.oM.Geometry.SettingOut.Grid to a Revit Grid or MultiSegmentGrid.")]
+        [Input("grid", "BH.oM.Geometry.SettingOut.Grid to be converted.")]
+        [Input("document", "Revit document, in which the output of the convert will be created.")]
+        [Input("settings", "Revit adapter settings to be used while performing the convert.")]
+        [Input("refObjects", "Optional, a collection of objects already processed in the current adapter action, stored to avoid processing the same object more than once.")]
+        [Output("grid", "Revit Grid or MultiSegmentGrid resulting from converting the input BH.oM.Geometry.SettingOut.Grid.")]
```
Comment (tiagogrossi) [inline Revit_Core_Engine/Convert/ToRevit.cs:48]: Same comments as above in this entire file. Many outputs don't match the output type.

**Review by tiagogrossi — COMMENTED** [review]
Minor comments really, but it seems there's quite a lot of outputs names that don't match the actual class output type. Is that intentional? Personally I'd rather have them equal for better clarity, what do you think?

**Review by tiagogrossi — APPROVED** [review]
Disregard my previous comments, I got confused. LGTM

**Comment by pawelbaran** [comment]
@BHoMBot check core
@BHoMBot check installer
@BHoMBot check null-handling
@BHoMBot check serialisation
@BHoMBot check versioning
@BHoMBot check ready-to-merge

**Pull request merged**